ipa-core/src/helpers/buffers/ordering_sender.rs

```diff
@@ -27,28 +25,17 @@ use crate::{
 /// The operating state for an `OrderingSender`.
 struct State {
     /// A store of bytes to write into.
-    buf: Vec<u8>,
-    /// The portion of the buffer that is marked "spare". Once `written + spare` is greater than
-    /// the buffer capacity, data is available to the stream. May be zero if messages are uniformly
-    /// sized and that size divides the buffer capacity.
-    spare: usize,
-    /// How many bytes have been written and are available.
-    written: usize,
-    /// The sender is closed.
-    closed: bool,
+    buf: CircularBuf,
     /// An entity to wake when the buffer is read from.
     write_ready: Option<Waker>,
     /// Another entity to wake when the buffer is read from.
     stream_ready: Option<Waker>,
 }
 
 impl State {
-    fn new(capacity: NonZeroUsize, spare: usize) -> Self {
+    fn new(capacity: usize, write_size: usize, read_threshold: usize) -> Self {
```

Comment:
should be write_size and everything else is a multiplier of write_size

Reply:
I looked at the code and I think it may be easier to leave it as is. The math behind calculating read_size and capacity is quite complex and needs to take into account the total records, active_work and read_size config parameter.

It is much easier to test this math by just validating the `SendChannelConfig`